### PR TITLE
Change default resource requirements for config reloader

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

~~Related to what was discussed in #5446, this PR changes the default resource requirements, but only for PrometheusAgent pods.~~

~~The number I've chosen is random, we probably want to discuss the final number~~

After discussing on the #prometheus-operator-dev Slack channel and gathering numbers from different production environments, we have good confidence that decreasing the CPU request from 100m to 10m isn't going to harm users.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [X] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
[CHANGE] Default CPU requirements of config-reloader set to 10m
```

### CODE REVIEW

**Review by JoaoBraveCoding — COMMENTED** [review]
Currently, I see this problem in the current implementation, as an alternative WDYT of leveraging the `options ...ReloaderOption` parameter in `CreateConfigReloader` and creating something like `SetCPUResources` and this would only be called by the agent code.

File: pkg/operator/config_reloader.go [hunk 1]
```diff
@@ -205,11 +205,6 @@ func CreateConfigReloader(name string, options ...ReloaderOption) v1.Container {
 		args = append(args, fmt.Sprintf("--log-format=%s", configReloader.logFormat))
 	}
 
-	resources := v1.ResourceRequirements{
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply pkg/operator/config_reloader.go:208]: What statement are you referring to? I'm not sure I follow 🤔

**Comment by ArthurSens** [comment]
> Currently, I see this ([#5539 (comment)](https://github.com/prometheus-operator/prometheus-operator/pull/5539#discussion_r1186257846)) problem in the current implementation, as an alternative WDYT of leveraging the `options ...ReloaderOption` parameter in `CreateConfigReloader` and creating something like `SetCPUResources` and this would only be called by the agent code.

First time doing anything on the configReloader side, I missed this strategy with ReloaderOption :)

Your idea seems a lot cleaner, let me check if I can get it working

File: pkg/prometheus/agent/statefulset.go [hunk 2]
```diff
@@ -481,3 +484,16 @@ func appendAgentArgs(promArgs []monitoringv1.Argument, cg *prompkg.ConfigGenerat
 
 	return promArgs
 }
+
+func getConfigReloaderDefaultCPU() v1.ResourceRequirements {
+	return v1.ResourceRequirements{
+		Requests: v1.ResourceList{
+			v1.ResourceCPU:    resource.MustParse("20m"),
```
[the remark this answers is not in the document]
Comment (ArthurSens) [reply pkg/prometheus/agent/statefulset.go:491]: I guess we could wait for the evidence to decide :P

**Comment by ArthurSens** [comment]
I was trying to figure out a way of getting rid of the function `getConfigReloaderDefaultResources` that was declared several times by just replacing it with a constant in each package, I think it would be cleaner. I hit a few import cycle errors though, then gave up on the idea 😅

**Comment by ArthurSens** [comment]
Default value for configReloader updated to 10 milicores in Agent deployments

**Comment by simonpasquier** [comment]
After all, I wonder if we shouldn't change the [global default](https://github.com/prometheus-operator/prometheus-operator/blob/960a97f917bc8724155c652e41c955e7276c7a81/cmd/operator/main.go#L63) value from 100m to 10m for all? Given the metrics you and @xiu have shared over Slack, even large Prometheus setups don't see much CPU activity for the reloader sidecar.

**Comment by ArthurSens** [comment]
Yeah, I think the vast majority wouldn't be affected by the change.


Would it be ok if we add a big warning in the next changelog?

**Comment by simonpasquier** [comment]
> Would it be ok if we add a big warning in the next changelog?

I'd be fine. IMHO changes like this are ok because we're not breaking the semantics of the API (as long as they are documented in the CHANGELOG).

**Comment by ArthurSens** [comment]
Updated the PR to change the default CPU requests for all

**Review by simonpasquier — APPROVED** [review]
lgtm

**Pull request merged**